File: src/main/java/com/example/checkoutplanner/entity/Schedule.java
package com.example.checkoutplanner.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.Data;

import java.util.List;

@Entity
@Data
public class Schedule {

    @Id
    private String date;

    private List<Long> shift_1;
    private List<Long> shift_2;
}

File: frontend/lib/models/schedule.dart
class Schedule {
  final String date;
  final List<int> shift_1;
  final List<int> shift_2;

  Schedule({
    required this.date,
    required this.shift_1,
    required this.shift_2,
  });

  factory Schedule.fromJson(Map<String, dynamic> json) {
    return Schedule(
      date: json['date'] ?? "",
      shift_1: List<int>.from(json['shift_1']),
      shift_2: List<int>.from(json['shift_2']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'date': date,
      'shift_1': shift_1,
      'shift_2': shift_2,
    };
  }
}
